src/parser/internal.ts

```diff
@@ -44,14 +47,47 @@ function parseInline(
   }
 }
 
-function parseParagraph(element: md.Paragraph): notion.ParagraphBlock {
+function parseParagraph(element: md.Paragraph): notion.Block {
+  // Paragraphs can contain inline elements that
+  // If a paragraph only contains one image child
```

Comment:
What is this comment trying to explain? The code below it looks like it handles invalid URLs but the comment mentions inline elements

Reply:
Sorry, it was a WIP comment, will fix.  The background is that while in markdown you can have images inline within paragraphs, in Notion you can't, they are always a parent block on their own, between paragraphs.

So my current fix for images is to:
- Look for paragraphs where the first element is an image.
- Reach inside, pull out the image, discard the rest - this assumes the paragraph only contains an image (which for most of our docs true).
- What then happens is that Notion is super strict on image urls being image urls of a certain type, so if you do create an image block, and for some reason Notion things the image url isn't valid, it rejects the *whole document* which isn't nice.  Hence the basic check here.
- On my wrapper library I am working on a processor that parses the images, finds them, puts them somewhere notion likes, and then replaces the URL, but that is a bit too specific for this module I think
- So if there is an error in the image url, I just leave the text of the url there so the user can see it was not processed.